here-naksha-storage-http/src/main/java/com/here/naksha/storage/http/connector/ConnectorInterfaceWriteExecute.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2017-2024 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.storage.http.connector;
+
+import static com.here.naksha.common.http.apis.ApiParamsConst.FEATURE_ID;
+import static com.here.naksha.lib.core.models.storage.ReadFeaturesProxyWrapper.ReadRequestType.GET_BY_ID;
+
+import com.here.naksha.lib.core.NakshaContext;
+import com.here.naksha.lib.core.exceptions.NoCursor;
+import com.here.naksha.lib.core.models.geojson.implementation.XyzFeature;
+import com.here.naksha.lib.core.models.geojson.implementation.namespaces.XyzNamespace;
+import com.here.naksha.lib.core.models.naksha.Space;
+import com.here.naksha.lib.core.models.payload.Event;
+import com.here.naksha.lib.core.models.payload.events.feature.*;
+import com.here.naksha.lib.core.models.storage.*;
+import com.here.naksha.lib.core.util.json.JsonSerializable;
+import com.here.naksha.storage.http.PrepareResult;
+import com.here.naksha.storage.http.RequestSender;
+import java.net.http.HttpResponse;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.jetbrains.annotations.NotNull;
+
+public class ConnectorInterfaceWriteExecute {
+
+  @NotNull
+  public static Result execute(NakshaContext context, WriteXyzFeatures request, RequestSender sender) {
+    String streamId = context.getStreamId();
+    String connectorSpaceName = request.getCollectionId();
+    Space connectorSpace = new Space(connectorSpaceName);
+
+    Event event = createCreateFeaturesEvent(context, request, sender, connectorSpaceName);
+
+    event.setSpace(connectorSpace);
+    event.setStreamId(streamId);
+
+    String jsonEvent = JsonSerializable.serialize(event);
+    HttpResponse<byte[]> httpResponse = sender.post(jsonEvent);
+
+    return PrepareResult.prepareWriteResult(httpResponse);
+  }
+
+  private static ModifyFeaturesEvent createCreateFeaturesEvent(
```

Comment:
Yes, this is what I moved to Tech Debt. But it should be done now, I can do it now

Reply:
Ah, ok I see it now. It need not be done immediately (as it is not blocker), however, my recommendation will be to address it before going for integration testing with the pilot team. I don't expect the fix to be complex, and just about adjusting how the loops are handled internally. It can be picked up after DELETE feature as well, if you prefer, so you get the overall picture before start making adjustment.

Reply:
Actually it turned out to be simpler than I thought.
Done